source/unified-test-format/unified-test-format.rst

```diff
@@ -369,6 +369,12 @@ The structure of this object is as follows:
   "sharded" topology, test runners MUST accept any type of sharded cluster (i.e.
   "sharded" implies "sharded-replicaset", but not vice versa).
 
+- ``serverless``: Optional string. Whether or not the test should be run on
+  serverless instances imitating sharded clusters. Valid values are "require",
+  "forbid", and "allow". If "require", the test MUST only be run on serverless
+  instances. If "forbid", the test MUST NOT be run on Serverless instances. If
+  omitted or "allow", this option has no effect.
```

Comment:
Can you confirm that serverless is entirely distinct from load balancers, in that a load balancer can front mongos _or_ serverless instances, but drivers can also connect directly to serverless without a load balancer?

I just want to make sure I'm understanding both projects correctly before we codify any `runOnRequirement` changes.

For context, let me quote @patrickfreed's previous comment (now hidden) from [this thread](https://github.com/mongodb/specifications/pull/944#discussion_r610796103):

> And yeah, I believe whatever restrictions apply to the serverless proxy probably also apply to the free-tier proxy, since I think the former is a descendant of / shares a lot of code with the latter, not sure about how tightly coupled they are though.

In #935, @divjotarora explained that drivers cannot infer what is behind a load balancer. This came up with respect with `useMultipleMongoses`. The solution there was to feed test runners two additional connection strings (one for an LB fronting a single mongos, and another for a LB fronting multiple mongoses), which can then be used as needed based on whether `useMultipleMongoses` is true or false. While this works in practice, it does invert the typical behavior for `useMultipleMongoses` in that drivers simply decide which connection string to use instead of inspecting the topology and possibly raising an error.

If serverless presents itself as mongos and cannot be differentiated, how is the test runner supposed to enforce the `serverless` option? I assume the test runner (either unified or legacy) will need to consult some explicit indicator (e.g. environment variable, configuration option) and decide based on that. If so, I think we need to clarify that. The test runners themselves need not be prescriptive beyond saying that it must be possible to inform them whether or not serverless is being used. The serverless testing doc itself can advise on a particular method for doing so.

Whatever mechanism is suggested will need to play nice with #935 in the event we ever test load balancers fronting serverless. #935 initially assumes that drivers will _only_ test load balancers with mongos, but I think we should consider how the `serverless` option will interact should we ever add load balancers fronting serverless to the test matrix. Expecting serverless usage to be communicated explicitly (e.g. environment variable or otherwise configuring the test runner) seems like a good way to avoid potential conflict there.

Reply:
By the time Serverless is GA, the only way to interact with it will be via a load balancer.